tests/unittests/ImageTest.cpp

```diff
@@ -55,6 +55,39 @@ TEST(Image, readBadImages) {
   ASSERT_FALSE(loadSuccess);
 }
 
+TEST(Image, readPngImageAndPreprocessWithAndWithoutInputTensor) {
+  auto image1 = readPngImageAndPreprocess(
+      "tests/images/imagenet/cat_285.png", ImageNormalizationMode::k0to1,
+      ImageChannelOrder::RGB, ImageLayout::NHWC, imagenetNormMean,
+      imagenetNormStd);
+  Tensor image2;
+  readPngImageAndPreprocess(image2, "tests/images/imagenet/cat_285.png",
+                            ImageNormalizationMode::k0to1,
+                            ImageChannelOrder::BGR, ImageLayout::NCHW,
+                            imagenetNormMean, imagenetNormStd);
+
+  // Test if the preprocess actually happened.
+  size_t imgHeight = image1.dims()[0];
+  size_t imgWidth = image1.dims()[1];
+  size_t numChannels = image1.dims()[2];
+
+  Tensor swizzled(image1.getType());
+  auto IH = image1.getHandle();
+  auto SH = swizzled.getHandle();
+  for (unsigned z = 0; z < numChannels; z++) {
```

Comment:
Nit: Counters are `unsigned` but bounds are `size_t`. Since the loop counter variables are used to index into Tensors they should probably be `size_t` as well.